io/plugins/eu.esdihumboldt.hale.io.jdbc/src/eu/esdihumboldt/hale/io/jdbc/JDBCTableCollection.java

```diff
@@ -304,8 +304,19 @@ public int size() {
 
 			return count;
 		} catch (SQLException e) {
-			log.warn("Could not determine query size", e);
-			return UNKNOWN_SIZE;
+			log.warn("Could not determine query size by count query");
+			try (Connection connection = createConnection()) {
+				Statement st = connection.createStatement();
```

Comment:
If the driver does not support it, calling `setMaxRows` will simply have no effect, but if the driver does support it, it will limit the amount of work performed on the server.